vortex-file/src/v2/file.rs

```diff
@@ -50,6 +52,69 @@ impl<I: IoDriver> VortexFile<I> {
 
     /// Performs a scan operation over the file.
     pub fn scan(&self, scan: Arc<Scan>) -> VortexResult<impl ArrayStream + 'static + use<'_, I>> {
+        self.scan_with_masks(
+            ArcIter::new(self.splits.clone())
+                .map(|row_range| RowMask::new_valid_between(row_range.start, row_range.end)),
+            scan,
+        )
+    }
+
+    /// Takes the given rows while also applying the filter and projection functions from a scan.
+    /// The row indices must be sorted.
+    pub fn take(
+        &self,
+        row_indices: Buffer<u64>,
```

Comment:
The V1 api took either a mask or indicies, maybe we should do the same? You might want to do a dense positional filter.